### ISSUE TITLE
Aborting worker is not working

### ISSUE THREAD

**Issue by saganshul** [owtf/owtf#762]
Aborting worker on worker panel is not working. 
Link to screenvideo:
https://www.dropbox.com/s/90atwivjpkhzfkq/out.mp4?dl=0
## Expected Behavior

It should stop the current job.
## Your Environment

- Installation method used: python install/install.py
- Version/branch used: Develop
- Operating System and version (like Kali, Debian, ArchLinux, etc): Ubuntu 16.04
- pip/setuptools version: 8.1.2
- sudo access: Yes

**Comment by DarKnight24** [comment]
@saganshul I am not able to reproduce the same.

**Comment by saganshul** [comment]
@DarKnight24 , try to test on plugin which takes time to run and try to abort the same.

**Comment by delta24** [comment]
@saganshul @DarKnight24 the request to kill the worker is sent successfully but sometimes the worker does not die immediately. Need to check the function in https://github.com/owtf/owtf/blob/develop/framework/plugin/worker_manager.py#L242.

**Comment by jack17529** [comment]
I am working on this issue can someone please tell me what's the meaning of - 
return True in [worker['busy'] for worker in self.workers]
https://github.com/owtf/owtf/blob/develop/framework/plugin/worker_manager.py#L141

**Comment by delta24** [comment]
@jack17529 `self.workers` is a list of workers. The line `for worker in self.workers` iterates over every worker in that list and gets the status of the worker using `worker['busy']`. The function returns `True` if any worker is 'busy'. 

The idiom used is a list comprehension - it is a syntactic way to write:
```
status = []
for worker in self.workers:
    status.append(worker['busy'])
return True in status
```

**Comment by jack17529** [comment]
Thank you @delta24

**Comment by jack17529** [comment]
Can someone explain meaning of "NA" in self.workers[k]["start_time"] = "NA"

https://github.com/owtf/owtf/blob/develop/framework/plugin/worker_manager.py#L119

**Comment by DePierre** [comment]
@jack17529 I would assume that 'NA' is a `start_time` error value used when the worker is idling. Since `start_time` cannot have any meaningful value if the worker is not working, 'NA' is used instead.

About your second question, you can get more meaning by running `git grep delete_worker`. You'll see that the `pseudo_index` refers to the index provided via the REST API. When the user clicks on the red cross to delete a worker, the URL `/api/workers/<id>/abort is fetched. This API will then call `delete_worker(<id>)` (therefore corresponding to `pseudo_index`, -1 to have the index starting from 0 and not 1). Then, as explained in the docstring, it ``Remove it from self.workers so that is is not restarted by manager cron``.

Do you need more information? We'll be happy to give you more details ;)

**Comment by ritwickdsouza** [comment]
Is this fixed ? I can't seem to produce the error. I ran a plugin which takes time and still was able to abort the worker

**Comment by jack17529** [comment]
@delta24  I have already looked all the code ,can't find any problem with code. But I think can be a problem related to the api used.What to do now ?

**Comment by DePierre** [comment]
@jack17529 I am not convinced that the API is at fault here and I would find it more likely that the issue is in our multiprocess system.

It is possible that the `worker_manager.py` is not the culprit here. Because a worker can run a shell command (to run `w3af` for instance), it is possible that when trying to send a SIGINT the worker does not receive it because `w3af` is freezing.

**Comment by viyatb** [comment]
Resolved by 2b2bd2214e538c5e1ed269fb57b75f91dd200e8c.
